src/main/java/com/ensao/gi5/lint/util/Utils.java

```diff
@@ -1,19 +1,143 @@
+     *
+     * It then uses the getFullyQualifiedName() method to retrieve the fully qualified name
+     * of the class or interface, which is stored in a String array called nameOfClass.
+     *
+     * Finally, the method returns the value stored in nameOfClass[0], which is the fully
+     * qualified name of the class or interface that contains the field declaration.
+     * The method is used to find the class name that contain a given field.
+     */
+    public static String findClassName(FieldDeclaration fieldDeclaration){
+        final String[] nameOfClass = new String[1];
+        fieldDeclaration.getParentNode().ifPresent(p->{
+            nameOfClass[0] = ((ClassOrInterfaceDeclaration) p).getFullyQualifiedName().orElse(null);
+        });
+        return nameOfClass[0];
+    }
+
+    /**
+     * @param enumDeclaration
+     * @return name of the enumeration
+     *
+     * This is a static method called findClassName that takes in a EnumConstantDeclaration
+     * object as its parameter. The method uses the getParentNode() method to access the
+     * parent node of the EnumConstantDeclaration, which should be an EnumDeclaration object.
+     *
+     * It then uses the getFullyQualifiedName() method to retrieve the fully qualified name
+     * of the enum, which is stored in a String array called nameOfEnum.
+     *
+     * Finally, the method returns the value stored in nameOfEnum[0], which is the fully qualified
+     * name of the enum that contains the EnumConstantDeclaration.
+     * The method is used to find the enum name that contain a given enum constant.
+     */
+    public static String findClassName(EnumConstantDeclaration enumDeclaration) {
+        final String[] nameOfEnum = new String[1];
+        enumDeclaration.getParentNode().ifPresent(p->{
+            nameOfEnum[0] = ((EnumDeclaration) p).getFullyQualifiedName().orElse(null);
+        });
+        return nameOfEnum[0];
+    }
+
+    /**
+     * @param constructorDeclaration
+     * @return name of class
+     *
+     * This method, findClassName(ConstructorDeclaration constructorDeclaration), is used to find the fully
+     * qualified name of the class that a given ConstructorDeclaration object belongs to.
+     *
+     * It uses the getParentNode() method of the ConstructorDeclaration object to access the parent node
+     * of the constructor, which is expected to be a ClassOrInterfaceDeclaration object.
+     * The fully qualified name of this class is then retrieved using the getFullyQualifiedName()
+     * method and stored in a local variable nameOfEnum. The method returns the value stored in nameOfEnum.
+     *
+     *  This method can be used in conjunction with the NumberOfParametersRule class to check for the
+     * number of parameters in a constructor and report violations if the number exceeds a certain threshold.
+     */
+    public static String findClassName(ConstructorDeclaration constructorDeclaration) {
```

Comment:
could be
```java
  public static String findClassName(Node node){
        return node.getParentNode()
                .filter(ClassOrInterfaceDeclaration.class::isInstance)
                .map(ClassOrInterfaceDeclaration.class::cast)
                .flatMap(ClassOrInterfaceDeclaration::getFullyQualifiedName)
                .orElse(null);
    }
```

this method replaces all the polymorphisms of the `findClassName`, because `Node` class is a parent of `EnumConstantDeclaration`, `ConstructorDeclaration`, `FieldDeclaration` ...